test/components/groupTests.ts

```diff
@@ -1,210 +1,208 @@
-    let c1 = new Plottable.Component;
-    let svg = TestMethods.generateSVG();
-    c1.renderTo(svg);
-    let group = new Plottable.Components.Group();
-    group.append(c1);
-    assert.isFalse((<Node> svg.node()).hasChildNodes(), "Component was detach()-ed");
-    svg.remove();
-  });
+    it("removes Components using remove()", () => {
+      let c0 = new Plottable.Component();
+      let c1 = new Plottable.Component();
+      let c2 = new Plottable.Component();
+      let componentGroup = new Plottable.Components.Group([c0, c1, c2]);
 
-  it("remove()", () => {
-    let c0 = new Plottable.Component();
-    let c1 = new Plottable.Component();
-    let c2 = new Plottable.Component();
-    let componentGroup = new Plottable.Components.Group([c0, c1, c2]);
+      componentGroup.remove(c1);
+      assert.deepEqual(componentGroup.components(), [c0, c2], "removing a Component respects the order of the remaining Components");
 
-    componentGroup.remove(c1);
-    assert.deepEqual(componentGroup.components(), [c0, c2], "removing a Component respects the order of the remaining Components");
+      let svg = TestMethods.generateSVG();
+      c1.renderTo(svg);
+      componentGroup.remove(c1);
+      assert.deepEqual(componentGroup.components(), [c0, c2],
+      "removing a Component not in the Group does not remove Components from the Group");
+      assert.strictEqual((<SVGElement> c1.content().node()).ownerSVGElement, svg.node(), "The Component not in the Group stayed put");
 
-    let svg = TestMethods.generateSVG();
-    c1.renderTo(svg);
-    componentGroup.remove(c1);
-    assert.deepEqual(componentGroup.components(), [c0, c2],
-    "removing a Component not in the Group does not remove Components from the Group");
-    assert.strictEqual((<SVGElement> c1.content().node()).ownerSVGElement, svg.node(), "The Component not in the Group stayed put");
+      svg.remove();
+    });
 
-    svg.remove();
+    it("checks for Components using has()", () => {
+      let c0 = new Plottable.Component();
+      let componentGroup = new Plottable.Components.Group([c0]);
+      assert.isTrue(componentGroup.has(c0), "correctly checks that Component is in the Group");
+      componentGroup.remove(c0);
+      assert.isFalse(componentGroup.has(c0), "correctly checks that Component is no longer in the Group");
+      componentGroup.append(c0);
+      assert.isTrue(componentGroup.has(c0), "correctly checks that Component is in the Group again");
+    });
   });
 
-  it("detach()-ing a Component that is in the Group removes it from the Group", () => {
-    let c0 = new Plottable.Component();
-    let componentGroup = new Plottable.Components.Group([c0]);
-    c0.detach();
-    assert.lengthOf(componentGroup.components(), 0, "Component is no longer in the Group");
-    assert.isNull(c0.parent(), "Component disconnected from Group");
-  });
+  describe("detach()ing constituent Components", () => {
```

Comment:
Describes generally start with a capital letter and do not contain method names. Hence
`Detaching constituent Components`